runbotics/runbotics-desktop/src/action/microsoft/excel/excel.types.ts

```diff
@@ -2,32 +2,26 @@ export type SessionIdentifier = string;
 export type WorksheetIdentifier = number | string;
 export type Platform = 'SharePoint' | 'OneDrive';
 export type ExcelCellValue = string | number | boolean;
-export interface Session {
+
+export interface OneDriveSessionInput {
     platform: Platform;
     sessionIdentifier: SessionIdentifier;
     worksheetIdentifier: WorksheetIdentifier;
-    workbookSessionInfo: WorkbookSessionInfo;
-    siteId?: string;
-    driveId?: string;
-    fileId?: string;
 }
 
-export interface SessionInput {
-    platform: Platform,
-    sessionIdentifier: SessionIdentifier,
-    worksheetIdentifier: WorksheetIdentifier,
-    siteRelativePath?: string,
+export interface SharePointSessionInput extends OneDriveSessionInput {
+    site?: string,
     list?: string
 }
 
-export interface SharePointSessionInput {
-    platform: Platform,
-    sessionIdentifier: SessionIdentifier,
-    worksheetIdentifier: WorksheetIdentifier,
-    siteRelativePath?: string,
-    list?: string
+export interface Session extends OneDriveSessionInput {
```

Comment:
imo Session should be top-level type and other types of sessions should inherit from it

Reply:
but in this case the hierarchy is reversed. "global" session object includes OneDrive and SharePoint session